src/orange/orangeinp/detail/DeMorganSimplifier.hh

```diff
@@ -17,6 +17,17 @@ namespace celeritas
 {
 namespace orangeinp
 {
+//---------------------------------------------------------------------------//
+/*!
+ * Result of a DeMorgan simplification.
+ */
+struct DeMorganSimplifierResult
```

Comment:
This could be a general "simplification" mapping. Perhaps it should be moved to CsgTree itself (imagine a function that runs "simplify" on the tree itself)?

Reply:
`CsgTree::simplify` only updates a single node, do we really want a vector with `tree.size()` elements for such a case?